### DISCUSSION TITLE
Use Usertable from mysql database

### DISCUSSION THREAD

**Discussion by lvsbird** [Budibase/budibase#2733] [Help] [answered]
Hi there, 
I'm almost new to budibase and trying some features to evaluate my business case.
I have a question:
I connected a mysql database to my created app.
Within the database I've got a user table which I want to use for authentication.
Is there any possibility to get this working?

Thank you!

**Comment by mjashanks** [comment, marked as answer]
Hi @lvsbird. This will not be possible with Budibase. Budibase has in-built user management. We support OpenID for SSO, so if you can get some OpenID service over the top of your database, then you can do it this way.

I'd like to understand more about your use case though. You should still be able to link your Budibase users to your database users - but Budibase will always handle the authentication, that's not going to change.

**Reply by lvsbird** [reply]
Hi @mjashanks .
Thank you for the fast reply :)
I'll try to explain my use case.
I've got a customer which has data for his clients.
My customer gets a login based on budibase credentials but his clients, which are in the table of the mysql database, should have a login too to get access to their personal data depending on the entries of the mysql database.

So far, I can manually create credentials for my customer, but it's impossible to create a login for each client (over 100)...

**Reply by mjashanks** [reply]
You could use the Budibase API & a script to seed Budibase with your logins. If the users already have hashed passwords stored in your MySQL db, then they would have to get a new password. 

Our API is undocumented at the moment, so there would be a bit of poking around to do, to understand what you need to do here. But - presumably, it's a one time job.
